extensions/amp-jwplayer/1.0/amp-jwplayer.js

```diff
@@ -0,0 +1,117 @@
+  getConsentPolicyState,
+} from '../../../src/consent';
+
+/** @const {string} */
+const TAG = 'amp-jwplayer';
+
+class AmpJwplayer extends BaseElement {
+  /** @override */
+  init() {
+    super.init();
+
+    const consentPolicy = this.getConsentPolicy();
+    if (consentPolicy) {
+      this.getConsentInfo().then((consentInfo) => {
+        const policyState = consentInfo[0];
+        const policyInfo = consentInfo[1];
+        const policyMetadata = consentInfo[2];
+        this.mutateProps(
+          dict({
+            'consentParams': {
+              'policyState': policyState,
+              'policyInfo': policyInfo,
+              'policyMetadata': policyMetadata,
+            },
+          })
+        );
+      });
+    }
+
+    return dict({
+      'queryParams': this.mergeQueryParams(
+        getDataParamsFromAttributes(this.element, null, /^playerParam(.+)/),
+        this.element.getAttribute('data-player-querystring')
+      ),
+      'contentSearch': this.getContextualSearch(
+        this.element.getAttribute('data-content-search')
+      ),
+    });
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    userAssert(
+      isExperimentOn(this.win, 'bento') ||
+        isExperimentOn(this.win, 'bento-jwplayer'),
+      'expected global "bento" or specific "bento-jwplayer" experiment to be enabled'
+    );
+    return super.isLayoutSupported(layout);
+  }
+
+  /**
+   * @param {string} policy
+   * @return {Promise}
+   */
+  getConsentInfo(policy) {
+    return Promise.all([
+      getConsentPolicyState(this.element, policy),
+      getConsentPolicyInfo(this.element, policy),
+      getConsentMetadata(this.element, policy),
+    ]);
+  }
+
+  /**
+   * Gets contextual search value
+   * @param {string} searchVal
+   * @return {string} contextual search value
+   */
+  getContextualSearch(searchVal) {
+    if (searchVal === '__CONTEXTUAL__') {
+      const context = this.element.ownerDocument.head;
+      const ogTitleElement = context.querySelector('meta[property="og:title"]');
+      const ogTitle = ogTitleElement
+        ? ogTitleElement.getAttribute('content')
+        : null;
+      const title = (context.querySelector('title') || {}).textContent;
+      return ogTitle || title || '';
+    }
+    return searchVal;
+  }
+
+  /**
+   * Merges query params from multiple sources into 1 string
+   * @param {Object} playerParams
+   * @param {string} playerQueryString
+   * @return {string}
+   */
+  mergeQueryParams(playerParams, playerQueryString) {
+    const addEntries = (acc, entries) => {
+      for (const entry of entries) {
+        acc[entry[0]] = entry[1];
+      }
+    };
+    const p1 = new URLSearchParams(playerParams || '');
+    const p2 = new URLSearchParams(playerQueryString || '');
+    const params = {};
+    addEntries(params, p1.entries());
+    addEntries(params, p2.entries());
+    return params;
+  }
+}
+
+AMP.extension(TAG, '1.0', (AMP) => {
+  AMP.registerElement(TAG, AmpJwplayer, CSS);
```

Comment:
recommending removing register css for reasons above
```suggestion
  AMP.registerElement(TAG, AmpJwplayer);
```